run/websockets/test/system.test.js

```diff
@@ -123,12 +123,23 @@ describe('End-to-End Tests', () => {
     await new Promise(resolve => setTimeout(resolve, 100));
 
     // Confirm message
-    const list = await browserPage.$('#messages li');
-    const itemText = await browserPage.evaluate(el => el.textContent, list);
-    assert.strictEqual(itemText.trim(), 'Sundar: Welcome!');
+    let itemText;
+    for (let i = 0; i < 5; i++) {
+      itemText = await browserPage.evaluate(() => {
+        return document.querySelector('#messages li');
+      });
+      if (itemText) return;
+      sleep(i * 1000); // Linear delay
```

Comment:
I think  you want to `await sleep(i * 1000);` here, otherwise the loop won't block.

Reply:
Oops. Thanks!